File: src/paquetegranadero/RobotStrategy.java
package paquetegranadero;

import robocode.JuniorRobot;

public interface RobotStrategy {


    void run(JuniorRobot robot);
    /**
     * onScannedRobot: What to do when you see another robot
     */
    void onScannedRobot(JuniorRobot robot);

    /**
     * onHitByBullet: What to do when you're hit by a bullet
     */
    void onHitByBullet(JuniorRobot robot);

    /**
     * onHitWall: What to do when you hit a wall
     */
    void onHitWall(JuniorRobot robot);

    void onHitRobot(JuniorRobot robot);
}

File: src/paquetegranadero/RobotStrategy.kt
package paquetegranadero

import robocode.JuniorRobot

interface RobotStrategy {

    fun run(robot : JuniorRobot)

    fun onScannedRobot(robot : JuniorRobot)

    fun onHitWall(robot : JuniorRobot)

    fun onHitByBullet(robot : JuniorRobot)

    fun onHitRobot(robot : JuniorRobot)

}